src/_pytest/logging.py

```diff
@@ -431,13 +436,48 @@ def clear(self) -> None:
         """Reset the list of log records and the captured log text."""
         self.handler.reset()
 
+    def force_enable_logging(
+        self, level: Union[int, str], logger_obj: logging.Logger
+    ) -> int:
+        """Enable the desired logging level if the level was disabled.
+
+        Only enables logging levels equal too and higher than the requested ``level``.
+
+        Does nothing if the desired ``level`` wasn't disabled.
+
+        :param Union[int, str] level:
+            The logger level caplog should capture.
+            All logging is enabled if a non-standard logging level string is supplied.
+            Valid level strings are in :data:`logging._nameToLevel`.
+        :param Logger logger_obj: The logger object to check.
+
+        :return int: The original disabled logging level.
+        """
+        original_disable_level: int = logger_obj.manager.disable  # type: ignore[attr-defined]
+
+        if isinstance(level, str):
+            level = logging.getLevelName(level)
+
+        if not isinstance(level, int):
+            logging.disable(logging.NOTSET)
+            return original_disable_level
+        if logger_obj.isEnabledFor(level):
+            return original_disable_level
+
+        disable_level = max(level - 10, logging.NOTSET)
+        logging.disable(disable_level)
+
+        return original_disable_level
```

Comment:
I’d recommend using `elif` instead of `return` and consolidating the three `return` statements to one.

```suggestion
        if not isinstance(level, int):
            logging.disable(logging.NOTSET)
        elif not logger_obj.isEnabledFor(level):
            # Each level is `10` away from other levels.
            # https://docs.python.org/3/library/logging.html#logging-levels
            disable_level = max(level - 10, logging.NOTSET)
            logging.disable(disable_level)        
        return original_disable_level
```

Is the logic still correct there?

Reply:
I am partial to return early, return often, but if this is the preferred style then I'll update my code to match. I'll make the change and make sure it still works.

Reply:
fixed